src/main/java/com/github/manolo8/darkbot/gui/titlebar/BackpageTask.java

```diff
@@ -0,0 +1,264 @@
+package com.github.manolo8.darkbot.gui.titlebar;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.extensions.util.Version;
+import com.github.manolo8.darkbot.gui.utils.Popups;
+import com.github.manolo8.darkbot.utils.FileUtils;
+import com.github.manolo8.darkbot.utils.I18n;
+import com.github.manolo8.darkbot.utils.OSUtil;
+import com.github.manolo8.darkbot.utils.Time;
+import com.github.manolo8.darkbot.utils.http.Http;
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import com.google.gson.annotations.SerializedName;
+import eu.darkbot.util.Timer;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.JButton;
+import javax.swing.JOptionPane;
+import javax.swing.JProgressBar;
+import javax.swing.SwingUtilities;
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class BackpageTask extends Thread {
+    private static final Path BACKPAGE_PATH = OSUtil.getDataPath("backpage");
+    private static final Path VERSION_PATH = BACKPAGE_PATH.resolve(".version");
+
+    private static final String EXECUTABLE_NAME = "dark_backpage";
+    private static final String RELEASE_URL = "https://api.github.com/repos/darkbot-reloaded/DarkBackpage/releases/latest";
+
+    private static final Timer VERSION_CHECK_TIMER = Timer.get(Time.HOUR * 12);
+
+    private final Main main;
+    private final BackpageButton button;
+
+    public BackpageTask(Main main, BackpageButton button) {
+        setDaemon(true);
+        this.main = main;
+        this.button = button;
+    }
+
+    @Override
+    public void run() {
+        if (main.backpage.isInstanceValid()) { // open backpage even if sid is KO
+            try {
+                if (canRun()) {
+                    new ProcessBuilder(BACKPAGE_PATH.resolve(EXECUTABLE_NAME).toAbsolutePath().toString(),
+                            "--sid", main.backpage.getSid(),
+                            "--url", main.backpage.getInstanceURI().toString())
+                            .start().waitFor();
+                }
+            } catch (InterruptedException | IOException e) {
+                e.printStackTrace();
+            }
+        } // inform user that instance/sid is not valid?
+        SwingUtilities.invokeLater(() -> button.setEnabled(true));
```

Comment:
given the amount of invoke-later's i'm seeing, consider using a SwingWorker instead